content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+    ShaderResourceGroupSemantic slot4 { FrequencyId = 4; };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON. Because --use-spaces is ON, SRG1 will be bound to Register Space 0.
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        Texture2D<float4>        m_texSRV_bounded[3];  // Will be bound to t2 register... Until t4 (inclusive).
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t5 register, and will take over ALL6+ for Register Space 0.
+        float m_value; // Declaring this variable, of Fundamental type, after an Unbounded Array is ok, as it is not a resource.
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON. Because --use-spaces is ON, SRG2 will be bound to Register Space 1.
+    ShaderResourceGroup SRG2 : slot2
+    {
+        RWTexture2D<float4>      m_texUAV_bounded[5];  // Will be bound to u1 register (Because The ConstantBuffer reserved for SRG2 will be bound to b0 register)...     Until u5 (inclusive).
+    	RWTexture2D<float4>      m_texUAV_unbounded[]; // Will be bound to u6 register, and will take over ALL7+ for Register Space 1.
+        bool m_boolValue; // Declaring this variable, of Fundamental type, after an Unbounded Array is ok, as it is not a resource.
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON. Because --use-spaces is ON, SRG3 will be bound to Register Space 2.
+    ShaderResourceGroup SRG3 : slot3
+    {
+        Sampler                  m_sampler1_bounded[7]; // Will be bound to s1 register (Because The ConstantBuffer reserved for SRG3 will be bound to b0 register)...     Until s7 (inclusive).
+        Sampler                  m_sampler2_bounded[3]; // Will be bound to s8 register... Until s10 (inclusive).
+        Sampler                  m_sampler_unbounded[]; // Will be bound to s11 register, and will take over ALL12+ for Register Space 2.
+        float3x3                 m_matrix; // Declaring this variable, of Fundamental type, after an Unbounded Array is ok, as it is not a resource.
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON. Because --use-spaces is ON, SRG4 will be bound to Register Space 3.
+    ShaderResourceGroup SRG4 : slot4
+    {
+        ConstantBuffer<MyStruct> m_cb_array[2]; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG4 will be bound to b0 register)... Until     b2 (inclusive).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b3 register.
+        ConstantBuffer<MyStruct> m_cb_unbounded[]; // Will be bound to b4 register, and will take over ALL5+ for Register Space 3.
+        int m_intValue; // Declaring this variable, of Fundamental type, after an Unbounded Array is ok, as it is not a resource.
+    };
```

Comment:
```suggestion
ShaderResourceGroupSemantic slot1 { FrequencyId = 1; };
ShaderResourceGroupSemantic slot2 { FrequencyId = 2; };
ShaderResourceGroupSemantic slot3 { FrequencyId = 3; };
ShaderResourceGroupSemantic slot4 { FrequencyId = 4; };

struct MyStruct
{
    float3 m_a;
    float3 m_b;
};

// For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
// SRG1 binds to register space 0. The ConstantBuffer reserved for SRG1 binds to t0 register. 
ShaderResourceGroup SRG1 : slot1
{
    Texture2D<float4>        m_texSRV1;             // Binds to t1 register.
    Texture2D<float4>        m_texSRV_bounded[3];   // Binds from t2 to t4 register (inclusive).
    Texture2D<float4>        m_texSRV_unbounded[];  // Binds to t5 register, and takes over ALL6+ for register space 0.
    float m_value;                                  // Okay to declare a variable of fundamental type as it is not a resource.
};

// SRG2 binds to register space 1. The ConstantBuffer reserved for SRG2 binds to u0 register. 
ShaderResourceGroup SRG2 : slot2
{
    RWTexture2D<float4>      m_texUAV_bounded[5];   // Binds from u1 to u5 register (inclusive).
    RWTexture2D<float4>      m_texUAV_unbounded[];  // Binds to u6 register and takes over ALL7+ for register space 1.
    bool m_boolValue;                               // Okay to declare a variable of fundamental type as it is not a resource.
};

// SRG3 binds to register space 2. The ConstantBuffer reserved for SRG3 binds to s0 register. 
ShaderResourceGroup SRG3 : slot3
{
    Sampler                  m_sampler1_bounded[7]; // Binds from s1 to s7 register (inclusive).
    Sampler                  m_sampler2_bounded[3]; // Binds from s8 to s10 register (inclusive).
    Sampler                  m_sampler_unbounded[]; // Binds to s11 register, and takes over ALL12+ for register space 2.
    float3x3                 m_matrix;              // Okay to declare a variable of fundamental type as it is not a resource.
};

// SRG4 binds to register space 3. The ConstantBuffer reserved for SRG4 binds to b0 register. 
ShaderResourceGroup SRG4 : slot4
{
    ConstantBuffer<MyStruct> m_cb_array[2];     // Binds from b1 to b2 register.
    ConstantBuffer<MyStruct> m_cb2;             // Binds to b3 register.
    ConstantBuffer<MyStruct> m_cb_unbounded[];  // Binds to b4 register, and takes over ALL5+ for register space 3.
    int m_intValue;                             // Okay to declare a variable of fundamental type as it is not a resource.
};
```
* Moved comment: "(Because the ConstantBuffer reserved for….)" to simplify the comments for each resource declaration. 
* Remove indent. It's not necessary when defining within a code snippet (```cpp). This also allows longer comments to fit on the page.
* Simplified "will be bound to" to "binds to".
* Simplified "will take over" to "takes over"

* Moved comment: "//For Vulkan & Metal…" before SRG1, since it applies to all SRGs in this code snippet. Removes redundancy. 
* For SRG1: Copy/paste typo in the comment? It says that SRG1 is bound to b0, but each resource seems to be bound to tN. 
   * Same comment for SRG2 (uN register space) and SRG3 (sN register space). 
   * SRG4 is correct (bN register space)

Reply:
Fixed. BUT the ConstantBuffers for SRG1, SRG2, SRG3 and SRG3 all  bind to b0 register.